### ISSUE TITLE
await check when calling promise type

### ISSUE THREAD

**Issue by pankleks** [Microsoft/TypeScript#15581]
This is just suggestion, but it would allow to eliminate quite common mistake (at least in my experience).

```ts
async function foo() {
...
}

foo(); // show warning that async is called without await
await foo(); // ok
```

Obviously it should be warning only as many times this is intentional.
From my experience this kind of mistakes are hard to find and easy to make - that's why I think it would be really useful.

Thanks for consideration.

**Comment by jesseschalken** [comment]
#10381

**Comment by pankleks** [comment]
ok, thanks!
